src/lc2kFormat.ts

```diff
@@ -1,13 +1,83 @@
 import * as vscode from 'vscode';
+import {TextDecoder} from 'util';
 
 export class Lc2kFormatter implements vscode.DocumentRangeFormattingEditProvider, vscode.DocumentFormattingEditProvider {
   provideDocumentFormattingEdits(document: vscode.TextDocument, options: vscode.FormattingOptions,
                                  token: vscode.CancellationToken): vscode.ProviderResult<vscode.TextEdit[]> {
-    return [];
+    let fullRange = new vscode.Range(0, 0, document.lineCount, 0);
+    return this.provideDocumentRangeFormattingEdits(document, fullRange, options, token);
   }
 
+
   provideDocumentRangeFormattingEdits(document: vscode.TextDocument, range: vscode.Range, options: vscode.FormattingOptions,
                                       token: vscode.CancellationToken): vscode.ProviderResult<vscode.TextEdit[]> {
-    return [];
+    // console.log("Formatting");
+    const config = vscode.workspace.getConfiguration('lc2k');
+    const enable = config.get('format.enable');
+    if (!enable) {
+      return [];
+    }
+    const fillOffset = +config.get('format.filloffset');
+    const jalrOffset = +config.get('format.jalroffset');
+    const noopOffset = +config.get('format.noopoffset');
+    const addNewLine = config.get('format.addNewlineAtEOF');
+    // console.log("jalr offset: " + jalrOffset);
+    // console.log("jalr offset: " + fillOffset);
+
+    var edits = [];
+    var regex;
+    var newText;
+    var offset;
+    for (let i = 0; i < document.lineCount; i++) {
+      let line = document.lineAt(i);
+      let text = line.text;
+      var OpcodeRegex = /(?:[^\s]*\s+)(add|nor|nand|lw|sw|beq|jalr|noop|halt|\.fill)(?:.*)/g;
+
+      let opcode = text.replace(OpcodeRegex, "$1");
+      // console.log("opcode: " + opcode);
+
+      switch (opcode) {
+        case "add":
+        case "nor":
+        case "lw":
+        case "sw":
+        case "beq":
+          regex = /^([^\s]*)\s+(add|nor|nand|lw|sw|beq)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+(.*)?/g;
```

Comment:
The final `\s+` will make this fail to match lines that have no comment in them.
Maybe try ending with:
```javascript
(?:\s+(.*))?;
```

Reply:
fixed by having whitespace be an optional capture